exporters/dsexporter/dsexporter.go

```diff
@@ -3,37 +3,70 @@ package main
 import (
 	"fmt"
 	"net/http"
+	"time"
 
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
 type CustomCollector struct {
-	requestCounter prometheus.Counter
+	konfluxProbeSuccess *prometheus.GaugeVec
 }
 
 // Creating a new instance of CustomCollector.
 func NewCustomCollector() *CustomCollector {
 	return &CustomCollector{
-		requestCounter: prometheus.NewCounter(prometheus.CounterOpts{
-			Name: "request_count",
-			Help: "Number of requests handled by the handler",
-		}),
+		konfluxProbeSuccess: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "konflux_probe_success",
+			Help: "Availability of konflux service/component",
+		},
+		[]string{"service", "check"}),
 	}
 }
 
 // Describe method sends descriptions of the metrics to Prometheus.
 // When Prometheus scrapes the /metrics endpoint of the exporter,
 // it first calls the Describe method to get a description of all the metrics.
 func (e *CustomCollector) Describe(ch chan<- *prometheus.Desc) {
-	e.requestCounter.Describe(ch)
+	e.konfluxProbeSuccess.Describe(ch)
 }
 
 // Collect method sends the current values of the metrics to Prometheus.
 // After Prometheus understands what metrics are available (using the `Describe` method),
 // it then calls the `Collect` method to actually get the values of those metrics.
 func (e *CustomCollector) Collect(ch chan<- prometheus.Metric) {
-	e.requestCounter.Collect(ch)
+	e.konfluxProbeSuccess.Collect(ch)
+}
+
+func availabilityHandler(response http.ResponseWriter, req *http.Request, exporter *CustomCollector) {
+	service := req.URL.Query().Get("service")
+	check := req.URL.Query().Get("check")
+	availability := checkIfServiceUp(service)
+
+	exporter.konfluxProbeSuccess.WithLabelValues(service, check).Set(availability)
+	fmt.Fprintf(response, "Service: %s, Check Name: %s, Availability: %f", service, check, availability)
+}
+
+func checkIfServiceUp(service string) float64 {
```

Comment:
```suggestion
func checkIfDependencyUp(url string) float64 {
```

Using service in the name or args might cause confusion with the metric's `service` label.

We should also pass in the full URL and not make any assumptions about what to query. Allowing the caller to pass an arbitrary host string opens this up to potential abuse. You can maintain a mapping somewhere for `check` to URL.

Reply:
Updated the code